Include/internal/pycore_pyerrors.h

```diff
@@ -24,16 +24,10 @@ static inline PyObject* _PyErr_Occurred(PyThreadState *tstate)
 
 static inline void _PyErr_ClearExcState(_PyErr_StackItem *exc_state)
 {
-    PyObject *t, *v, *tb;
-    t = exc_state->exc_type;
+    PyObject *v;
```

Comment:
If only one variable is left, can we omit it altogether?

```c
static inline void _PyErr_ClearExcState(_PyErr_StackItem *exc_state)
{
    Py_XDECREF(exc_state->exc_value);
    exc_state->exc_value = NULL;
}
```

Reply:
No it's a linked list, it has a next pointer.

Reply:
If I understand it correctly, the function does not treat it so. It withdraws a pointer to the whole *exc_value* leaving NULL behind, then releases *exc_value*:

https://github.com/python/cpython/blob/a72556e7f7026c4788006ce9d2a50538292bc47f/Include/internal/pycore_pyerrors.h#L25-L31

With simplified identifiers (*exc_state->exc_value* replaced with *a*):

```c
PyObject *v = a;
a = NULL;
Py_XDECREF(v);
```

Py_XDECREF also has no list-related magic under the hood:

https://github.com/python/cpython/blob/8319114feedd2a5b77378bba24eb9fb2689c5033/Include/object.h#L581-L588

https://github.com/python/cpython/blob/8319114feedd2a5b77378bba24eb9fb2689c5033/Include/object.h#L112

Reply:
I probably don't understand what you suggested to omit.

Reply:
@arhadthedev's suggestion is to remove the temporary variable `v`, but his recommended change is incorrect as the decref is done before NULLing the value.

You could change
```
PyObject *v;
v = exc_state->exc_value;
exc_state->exc_value = NULL;
Py_XDECREF(v);
```
to
```
Py_CLEAR(exc_state->exc_value);
```
if you want.